packages/eslint-plugin/src/rules/restrict-template-expressions.ts

```diff
@@ -175,7 +122,12 @@ export default createRule<Options, MessageId>({
           return innerType.types.some(rec);
         }
 
-        return predicate(innerType);
+        return (
+          isTypeFlagSet(innerType, ts.TypeFlags.StringLike) ||
+          optionsArr.some(
+            ({ option, fn }) => options[option] && fn(innerType, checker, rec),
```

Comment:
> `options[option]`

[Refactor] Maybe it'd make sense to generate this list within the rule, so a bunch of truthiness checks don't need to happen each time a template expression is looked at?